packages/frontend/src/providers/ExplorerProvider.tsx

```diff
@@ -578,27 +588,29 @@ export const ExplorerProvider: FC = ({ children }) => {
         });
     }, []);
 
-    const pristineState = useMemo(
-        () => ({
-            ...pristineReducerState,
-            activeFields: pristineActiveFields,
-            isValidQuery: pristineIsValidQuery,
-        }),
-        [pristineActiveFields, pristineIsValidQuery, pristineReducerState],
+    const state = useMemo(
+        () => ({ ...reducerState, activeFields, isValidQuery }),
+        [reducerState, activeFields, isValidQuery],
     );
-    const queryResults = useQueryResults(pristineState);
+    const queryResults = useQueryResults(state);
+
+    // Fetch query results after state update
+    const { mutate } = queryResults;
+    useEffect(() => {
+        if (state.shouldFetchResults) {
+            mutate();
+            dispatch({
+                type: ActionType.RESET_SHOULD_FETCH_RESULTS,
```

Comment:
I read this as "The reset should fetch results" rather than "Reset the field 'should fetch results'"

How about "SET_FETCH_RESULTS_FALSE"